### ISSUE TITLE
Need new ApplicationTokenProvider API to support authenticate by Subject Name and Issuer of cert

### ISSUE THREAD

**Issue by JackLiang5** [Azure/azure-sdk-for-net#5206]
When using Azure DNS client, 
```
var clientCreds = await ApplicationTokenProvider.LoginSilentWithCertificateAsync(domain, new ClientAssertionCertificate(clientId, cert)).ConfigureAwait(false);
var dnsClient = new DnsManagementClient(clientCreds);
```

To support authenticate by Subject Name and Issuer of cert, we need a new API like:
`ApplicationTokenProvider.LoginSilentWithCertificateAsync(string domain, IClientAssertionCertificate certificate, bool sendX5c);
`

Thereinto, LoginSilentWithCertificateAsync will forward argument sendX5c to AcquireTokenAsync.
`AuthenticationContext.AcquireTokenAsync(string resource, IClientAssertionCertificate clientCertificate, bool sendX5c)`

For more detail, please ref to https://aadwiki.windows-int.net/index.php?title=Subject_Name_and_Issuer_Authentication_-_Advanced_Administrator_Guidance

**Comment by m-mertz** [comment]
We're also blocked by this. Is there an ETA?

**Comment by jsquire** [comment]
Given the age of this issue, the linked guidance for MSAL use, and the new approach to identity in the current generation of libraries, I'm going to close this out.   If there is still a need for this feature with `Azure.Identity`, please open a new issue for discussion.
